src/prefect/_internal/compatibility/experimental.py

```diff
@@ -229,6 +229,9 @@ def __init__(__pydantic_self__, **data: Any) -> None:
             # Perform warning check
             if name in data.keys() and when(data[name]):
                 experimental_check()
+            field = __pydantic_self__.__fields__.get(name)
+            if field is not None and not _opt_in_setting_for_group(group):
+                field.field_info.extra["experimental"] = True
```

Comment:
Maybe we should always set `experimental` to True then have `experiment-enabled` as a separate key? I'd be confused as-is. Checking here seems a little more performant than checking at `dict` but is it worth it?

Reply:
I included `experimental-group` so that we can check in other places if need for opt-in or warning. Does that seem reasonable?

Reply:
Sounds good to me!